language/tools/read-write-set/src/dynamic_analysis.rs

```diff
@@ -82,18 +82,22 @@ impl ConcretizedFormals {
     fn concretize_secondary_indexes(
         self,
         blockchain_view: &impl MoveResolver,
-        env: &GlobalEnv,
-    ) -> ConcretizedSecondaryIndexes {
+        fun_env: &FunctionEnv,
+    ) -> Result<ConcretizedSecondaryIndexes> {
         // TODO: check if there are no secondary indexes and return accesses if so
         let annotator = MoveValueAnnotator::new(blockchain_view);
         let mut acc = AccessPathTrie::default();
-        // TODO: do not unwrap here. iter_paths doesn't support Result, so need to create an Iter instead
-        // of iter_paths or return a bool inside resolve_secondary instead of using ?
-        self.0.iter_paths(|access_path, access| {
-            Self::concretize_secondary_indexes_(&annotator, access_path, access, env, &mut acc)
-                .unwrap();
+        let results = self.0.map_paths(|access_path, access| {
```

Comment:
@sblackshear It's slightly out of scope for this PR but since I was looking at stuff around the dynamic analysis I also fixed this TODO. Let me know if you'd prefer this as a separate PR

Reply:
Nice!